bin/xbps-pkgdb/check_pkg_symlinks.c

```diff
@@ -53,6 +53,10 @@ check_pkg_symlinks(struct xbps_handle *xhp, const char *pkgname, void *arg)
 	xbps_object_t obj;
 	xbps_dictionary_t filesd = arg;
 	int rv = 0;
+	struct stat st;
+	mode_t mode;
+	uid_t uid;
+	gid_t gid;
```

Comment:
I would move those variables in the blocks where they are used. `st` into the for loop and `mode`, `uid`, `gid` into the `if`.